internal/ethapi/api.go

```diff
@@ -1000,24 +1006,50 @@ func DoEstimateGas(ctx context.Context, b Backend, args CallArgs, blockNrOrHash
 						revert = ret
 					}
 				}
-				return 0, estimateGasError{
+				return 0, nil, estimateGasError{
 					error:  "always failing transaction",
 					vmerr:  result.Err,
 					revert: revert,
 				}
 			}
 			// Otherwise, the specified gas cap is too low
-			return 0, estimateGasError{error: fmt.Sprintf("gas required exceeds allowance (%d)", cap)}
+			return 0, nil, estimateGasError{error: fmt.Sprintf("gas required exceeds allowance (%d)", cap)}
 		}
+		validState = prevS
 	}
-	return hexutil.Uint64(hi), nil
+	return hexutil.Uint64(hi), validState, nil
 }
 
-// EstimateGas returns an estimate of the amount of gas needed to execute the
-// given transaction against the current pending block.
-func (s *PublicBlockChainAPI) EstimateGas(ctx context.Context, args CallArgs) (hexutil.Uint64, error) {
+func (s *PublicBlockChainAPI) EstimateGas(ctx context.Context, argsInterface interface{}) (interface{}, error) {
```

Comment:
It's a little bit hacky to me. What about adding another API with `args []CallArgs` as the parameters?

Reply:
Yea that works for me, how about eth_estimateGases ? Or do you have another one in mind. I didn't want to make changes to the standard API that's why just decided to go with the current implementation

Reply:
Sounds good to me

Reply:
`estimateGases` sounds weird to me haha
I added `eth_estimateGasList`